Project/DAQ_System/Program/Src/Platform/CAN/Interfaces/ican.hpp

```diff
@@ -12,12 +12,30 @@
 #ifndef ICAN_H
 #define ICAN_H
 
+// Standard Libraries
+#include <cstdint>
+
 namespace platform {
 
 class ICan {
-    public:
-        virtual ~ICan() = default;
+public:
+	static constexpr uint8_t kMaxBytes = 8;
+
+	virtual ~ICan() = default;
+
+	virtual void Receive(uint8_t rx_buffer[kMaxBytes]) = 0;
+
+	virtual void Transmit(uint8_t tx_buffer) = 0;
+
+	virtual bool MessageArrivedFlag() = 0;
+
+	virtual void ClearMessageArrivedFlag() = 0;
+
+	virtual uint32_t LatestCanId() = 0;
+
+	virtual void EnableInterruptMode() = 0;
 
+	virtual void DisableInterruptMode() = 0;
```

Comment:
This isn't relevant to what we are doing but having another function in the abstract interface where you can set custom ID masks for the CAN Controller filtering might be a nice addition to the abstract interface.

Reply:
Done. The user of the library is now expected to do the following:
- Provide the BxCAN peripheral from the library through the constructor
- Provide a list of CAN IDs
- Start the peripheral
- Configure the interrupt for the receive callback
- Enable the interrupt
- Profit